### PR TITLE
Aequatus generator enhancements

### PR DESCRIPTION
**Pull request by anilthanki**
New test data and changes in Aequatus format with added  information about reference, replaces test data with BRCA genes

### CODE REVIEW

**Comment by nsoranzo** [comment]
@anilthanki Can you fix the errors/warnings mentioned in https://travis-ci.org/TGAC/tgac-galaxytools/builds/110103025 (lines 174-183)?

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "visualisation/aequatus/static/aequatus-vis"]
+	path = visualisation/aequatus/static/aequatus-vis
+	url = https://github.com/anilthanki/aequatus-vis.git
```
Comment (nsoranzo) [inline .gitmodules]: I think this file should be only added in PR #15.

File: tools/aequatus/aequatus-generator.py [hunk 2]
```diff
@@ -35,15 +53,58 @@ def gene_to_json(fname):
...
+
+    aequatus_dict['ref'] = refGene
+    aequatus_dict['protein_id'] = refProtein
+    aequatus_dict['transcript_id'] = refTrancript
+    aequatus_dict['version'] = version
+    
     return aequatus_dict
 
+def parseGeneJSON(obj, id):
+    global refProtein
```
Comment (nsoranzo) [inline tools/aequatus/aequatus-generator.py]: `refProtein` is not modified in this function, do you mean `refTrancript`?

P.S.: global variables are considered evil in programming. All mutable objects (lists, dictionaries, class objects,...) are passed by reference in Python, so you can just add all these variables as function parameters.

File: tools/aequatus/aequatus-generator.py [hunk 3]
```diff
@@ -35,15 +54,61 @@ def gene_to_json(fname):
 
 
 def join_json(tree_dict, gene_dict, cigar_dict):
+    global version
+
```
Comment (nsoranzo) [inline tools/aequatus/aequatus-generator.py]: This is not needed because you do not modify `version`.

File: tools/aequatus/aequatus-generator.py [hunk 4]
```diff
@@ -11,11 +19,22 @@ def write_json(aequatus_dict, outfile=None, sort_keys=False):
 
 
 def cigar_to_json(fname):
+    global refTrancript
```
Comment (nsoranzo) [inline tools/aequatus/aequatus-generator.py]: `refTrancript` is not used by this function.

File: tools/aequatus/aequatus-generator.py [hunk 5]
```diff
@@ -35,15 +54,61 @@ def gene_to_json(fname):
...
+                    return each["Translation"]["id"]
+
+
+def parseTree(obj):
+    global refProtein
+    global refGene
+    global refTrancript
+    if "children" not in obj:
+        jsonArrayGene.append(obj["id"]["accession"])
+        jsonArrayTranscript.append(obj["sequence"]["id"][0]["accession"])
```
Comment (nsoranzo) [inline tools/aequatus/aequatus-generator.py]: The `jsonArrayGene` and `jsonArrayTranscript` lists are not used after being created, can be deleted?

File: tools/aequatus/aequatus-generator.py [hunk 6]
```diff
@@ -35,15 +54,61 @@ def gene_to_json(fname):
...
+    for obj in gene_dict:
+        if "Transcript" in gene_dict[obj]:
+            for each in gene_dict[obj]["Transcript"]:
+                if each["id"] == id:
+                    return each["Translation"]["id"]
+
+
+def parseTree(obj):
+    global refProtein
+    global refGene
```
Comment (nsoranzo) [inline tools/aequatus/aequatus-generator.py]: `refGene` is not used by this function.

File: tools/aequatus/aequatus-generator.py [hunk 7]
```diff
@@ -35,15 +54,61 @@ def gene_to_json(fname):
...
+    global refTrancript
```
Comment (nsoranzo) [inline tools/aequatus/aequatus-generator.py]: `refTrancript` is not used by this function.

**Comment by nsoranzo** [comment]
@anilthanki I've added a few commits to your branch, can you check them, please?

**Comment by anilthanki** [comment]
Checked and tested in Galaxy it works.. please go ahead with merging if its ready

File: tools/aequatus/aequatus.xml [hunk 8]
```diff
@@ -41,7 +41,14 @@ $sort
             <param name="alignmentFile" ftype="tabular" value="cigar.tabular" />
             <param name="genesFile" ftype="json" value="gene.json" />
             <param name="sort" value="-s" />
-            <output name="outputFile" ftype="aequatus" file="test.aequatus" />
+            <output name="outputFile" ftype="aequatus" file="test1.aequatus" />
+        </test>
+        <test>
+            <param name="input_selector" value="json" />
+            <param name="treeFile" ftype="nhx" value="tree.json" />
```
Comment (nsoranzo) [inline tools/aequatus/aequatus.xml]: s/nhx/json/

**Pull request merged**